scipy/stats/stats.py

```diff
@@ -3999,6 +3996,51 @@ def pearsonr(x, y):
     >>> stats.pearsonr([1, 2, 3, 4, 5], [10, 9, 2.5, 6, 4])
     (-0.7426106572325057, 0.1505558088534455)
 
+    It is important to keep in mind that no correlation does not imply
+    independence unless (x, y) is Gaussian. Correlation can even be zero
+    when there is a very simple dependence structure: if x follows a
+    standard normal distribution, let y = abs(x). A simple check
+    confirms that the correlation is close to zero:
+
+    >>> x = stats.norm.rvs(size=500)
+    >>> y = np.abs(x)
+    >>> stats.pearsonr(x, y)
+    (-0.016172891856853524, 0.7182823678751942) # may vary
+
+    Indeed, since the expectation of x is zero, cov(x, y) = E[x*y]. This equals
+    E[x*abs(x)] which is zero by symmetry.
```

Comment:
It might just be six vs half-a-dozen, but I'll propose it anyway: put what you expect mathematically right after you've defined "x" and "y = abs(x)". Then, "Indeed, a simple check confirms..." 

Only change if that sounds good to you.

Reply:
i tried to simplify the description